src/patient-registration/field/address/address-field.component.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react';
+import styles from '../field.scss';
+import { Input } from '../../input/basic-input/input/input.component';
+import { getAddressTemplate } from '../../patient-registration.resource';
+
+export const AddressField: React.FC = () => {
+  const [addressFields, setAddressFields] = useState([]);
+  const [addressTemplate, setAddressTemplate] = useState('');
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    getAddressTemplate(abortController).then(({ data }) => {
+      setAddressTemplate(data.results[0].value);
+    });
+  }, []);
+
+  useEffect(() => {
+    const templateXmlDoc = parseString(addressTemplate);
+    const nameMappings = getTagAsDocument('nameMappings', templateXmlDoc);
+    const elementDefaults = getTagAsDocument('elementdefaults', templateXmlDoc);
+    const properties = nameMappings.getElementsByTagName('property');
+    const propertiesObj = Array.prototype.map.call(properties, (property: Element) => {
+      const name = property.getAttribute('name');
+      const labelText = property.getAttribute('value');
+      const value = getFieldValue(name, elementDefaults);
+      return {
+        id: name,
+        name,
+        labelText,
+        value,
+      };
+    });
+    setAddressFields(propertiesObj);
+  }, [addressTemplate]);
+
+  const parseString = (xmlDockAsString: string) => new DOMParser().parseFromString(xmlDockAsString, 'text/xml');
+
+  const getTagAsDocument = (tagName: string, template: XMLDocument) => {
```

Comment:
This could be moved outside of the component to prevent re-definition (and thus avoid linter complains as the one above). I don't see any captured variable justifying having it in the component.